src/analysis/bounded_variation.lean

```diff
@@ -627,6 +628,44 @@ begin
   exact comp_le_of_monotone_on _ ψ hψ ψts,
 end
 
+namespace set
+
+-- porting note: move to file `data.set.intervals.basic` once the port is over.
+lemma Icc_subsingleton_of_ge {α : Type*} [partial_order α] {a b : α} (h : b ≤ a) :
+  set.subsingleton (Icc a b) :=
+begin
+  rintros c ⟨ac,cb⟩ d ⟨ad,db⟩,
+  cases le_antisymm (cb.trans h) ac,
+  cases le_antisymm (db.trans h) ad,
+  refl,
+end
+
+end set
+
+lemma comp_eq_of_monotone_on_inter_Icc (f : α → E) {s : set α} {t : set β} (φ : β → α)
+  (hφ : monotone_on φ t) (φst : set.maps_to φ t s) (φsur : set.surj_on φ t s) {x y : β}
```

Comment:
can you `open set` to avoid writing `set.maps_to` and `set.Icc` and so on (in the whole file)?